### PR TITLE
refactor: `Enum.count` to `Enum.empty?`

### PR DESCRIPTION
**Pull request by mohammedzeglam-pg**
## Motivation

Enhance Enum usage could gain performance hits for example `!Enum.empty?(list)` has a constant time.

### Enhancements
Improve Enum usage

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

**Review by fedor-ivn — CHANGES_REQUESTED** [review]
Thanks for the PR. Please, add an entry to `CHANGELOG.md`.

File: apps/block_scout_web/lib/block_scout_web/notifier.ex [hunk 1]
```diff
@@ -433,7 +433,7 @@ defmodule BlockScoutWeb.Notifier do
   end
 
   defp broadcast_transactions_websocket_v2_inner(transactions, default_channel, event) do
-    if Enum.count(transactions) > 0 do
+    if !Enum.empty?(transactions) do
```
Comment (fedor-ivn) [inline apps/block_scout_web/lib/block_scout_web/notifier.ex:436]: I would suggest using `not` instead of `!`

Comment (mohammedzeglam-pg) [reply apps/block_scout_web/lib/block_scout_web/notifier.ex]: Thanks i updated

**Comment by mohammedzeglam-pg** [comment]
> Please fix [these](https://github.com/blockscout/blockscout/actions/runs/8342550957/job/22831118754#step:5:1) warnings (see refactoring opportunities). I guess it would involve just swapping the if branches and removing the negation in the condition statement.

Yes you are correct.

File: apps/block_scout_web/lib/block_scout_web/templates/layout/_topnav.html.eex [hunk 2]
```diff
@@ -161,19 +161,19 @@
...
           <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
-            <%= if Enum.count(main_nets) > 0 do %>
+            <%= if not Enum.empty?(main_nets) do %>
               <a class="dropdown-item header division">Mainnets</a>
               <%= for %{url: url, title: title} <- main_nets do %>
                 <a class="dropdown-item" href="<%= url %>"><%= title %></a>
               <% end %>
             <% end %>
-            <%= if Enum.count(test_nets) > 0 do %>
+            <%= if not Enum.empty?(test_nets) do %>
```
Comment (fedor-ivn) [inline apps/block_scout_web/lib/block_scout_web/templates/layout/_topnav.html.eex:170]: I guess it makes sense to also refactor such `not Enum.empty?(...)` conditions in the templates.

Comment (mohammedzeglam-pg) [reply apps/block_scout_web/lib/block_scout_web/templates/layout/_topnav.html.eex:170]: you mean use `unless`

Comment (fedor-ivn) [reply apps/block_scout_web/lib/block_scout_web/templates/layout/_topnav.html.eex]: This also an option

Comment (mohammedzeglam-pg) [reply apps/block_scout_web/lib/block_scout_web/templates/layout/_topnav.html.eex]: in template i refactor it to use `unless`

**Review by fedor-ivn — APPROVED** [review]

**Review by sl1depengwyn — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -70,6 +70,7 @@
 
 ### Chore
 
+- [#9666](https://github.com/blockscout/blockscout/pull/9666) - Improved performance by replacing `Enum.count(list) == 0` with `Enum.empty?(list)` for better readability and efficiency.
```
Comment (vbaranov) [inline CHANGELOG.md:73]: Sorry, for making a request about changelog again. Could you please remove CHANGELOG record? We recently migrated to semi-auto generation of changelog.

Comment (mohammedzeglam-pg) [reply CHANGELOG.md]: done

**Pull request merged**